Engine/src/texture.cpp

```diff
@@ -0,0 +1,225 @@
+#include <texture.h>
+#include <glad/glad.h>
+#include <timer.h>
+#include <math.h>
+
+namespace Slicer
+{
+Texture::Texture(const std::shared_ptr<ApplicationState>& state,
+                 std::shared_ptr<CoordinateSystem> parent)
+:Model(state)
+,mVAO(0)
+,mVerticesBO(0)
+,mVertices()
+,mTextureCoordsBO(0)
+,mTextureCoords()
+,mSliceBO(0)
+,mSlice()
+,mData()
+{
+    resetCS(std::shared_ptr<CoordinateSystem>(new CoordinateSystem(glm::mat4(1.0f), parent)));
+    initializeModel();
+    initializeMembers();
+}
+
+Texture::~Texture()
+{
+}
+
+void Texture::updateApplicationStateAtInit()
+{
+}
+
+void Texture::registerStateCallbacks()
+{
+}
+
+
+void Texture::initProgramPipeline()
+{
+    const std::string vsPath = DMRI_EXPLORER_BINARY_DIR + std::string("/shaders/texture_vert.glsl");
+    const std::string fsPath = DMRI_EXPLORER_BINARY_DIR + std::string("/shaders/texture_frag.glsl");
+    std::vector<GPU::ShaderProgram> shaders;
+
+    shaders.push_back(GPU::ShaderProgram(vsPath, GL_VERTEX_SHADER));
+    shaders.push_back(GPU::ShaderProgram(fsPath, GL_FRAGMENT_SHADER));
+
+    mProgramPipeline = GPU::ProgramPipeline(shaders);
+}
+
+void Texture::initializeMembers()
+{
+    const auto& image = mState->BackgroundImage.Get();
+
+    const int dimX = image.dims().x;
+    const int dimY = image.dims().y;
+    const int dimZ = image.dims().z;
+    const int nCoeffs = image.dims().w;
+
+    const double max = image.getMax();
+    
+    //Get image data
+    for(int k = 0; k < dimZ; ++k)
+    {
+        for(int j = 0; j < dimY; ++j)
+        {
+            for(int i = 0; i < dimX; ++i)
+            {
+                for(int l = 0; l < nCoeffs; ++l)
+                {
+                    mData.push_back(float(image.at(i,j,k,l))/max);
```

Comment:
Faire le calcul en double, puis caster en float.
```suggestion
                    mData.push_back(float(image.at(i,j,k,l)/max));
```